### PR TITLE
CLI for offload

### PR DESCRIPTION
**Pull request by ivankelly**
This patch adds two CLI commands for offload, one to trigger offload,
and another to check offload status.

Triggering offload requires a size-threshold argument, which is the
maximum number of bytes which should be retained locally. The prefix
of ledgers over this threshold will be offloaded.

```
pulsar-admin topics offload --size-threshold 104857600 persistent://public/default/topic1
pulsar-admin topics offload-status -w persistent://public/default/topic1
```

Master Issue: #1511

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 1]
```diff
@@ -614,6 +618,93 @@ void run() throws PulsarAdminException {
...
+        for (PersistentTopicInternalStats.LedgerInfo l : ledgers) {
+            suffixSize += l.size;
+            if (suffixSize >= sizeThreshold) {
+                return new MessageIdImpl(l.ledgerId, 0L, -1);
+            }
+        }
+        return null;
+    }
+
+    @Parameters(commandDescription = "Trigger offload data from a topic to long term storage")
```
Comment (lucperkins) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: "Trigger offload of data from a topic to long-term storage"

Comment (lucperkins) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: It also might help to include `(e.g. Amazon S3)` in the description. Up to you.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 2]
```diff
@@ -614,6 +618,93 @@ void run() throws PulsarAdminException {
...
+    private class Offload extends CliCommand {
+        @Parameter(names = { "-s", "--size-threshold" },
+                   description = "Max amount of data to keep in bookkeeper for topic", required = true)
```
Comment (lucperkins) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: "Max amount of data to keep in BookKeeper for the specified topic"

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 3]
```diff
@@ -614,6 +618,93 @@ void run() throws PulsarAdminException {
...
+                System.out.println("Nothing to offload");
+                return;
+            }
+
+            topics.triggerOffload(persistentTopic, messageId);
+            System.out.println("Offload triggered for " + persistentTopic + " for messages before " + messageId);
+        }
+    }
+
+    @Parameters(commandDescription = "Status of offloading on a topic")
```
Comment (lucperkins) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: "Check the status of data offloading from a a topic to long-term storage"

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java (hunk 1 above)
Comment (ivankelly) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: done

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java (hunk 2 above)
Comment (ivankelly) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: done

**Pull request merged**